pkg/splunk/enterprise/searchheadcluster.go

```diff
@@ -233,6 +247,67 @@ func ApplyShcSecret(mgr *searchHeadClusterPodManager, replicas int32, mock bool)
 				mgr.cr.Status.ShcSecretChanged = append(mgr.cr.Status.ShcSecretChanged, true)
 			}
 		}
+
+		// If admin secret is different from namespace scoped secret change it
+		if adminPwd != nsAdminSecret {
+			scopedLog.Info("admin password different from namespace scoped secret, changing admin password")
+			// If admin password already changed, ignore
+			if i < int32(len(mgr.cr.Status.AdminSecretChanged)) {
+				if mgr.cr.Status.AdminSecretChanged[i] {
+					continue
+				}
+			}
+
+			// Change admin password on splunk instance of pod
+			command := fmt.Sprintf("/opt/splunk/bin/splunk cmd splunkd rest --noauth POST /services/admin/users/admin 'password=%s'", nsAdminSecret)
+			_, _, err = splutil.PodExecCommand(mgr.c, shPodName, mgr.cr.GetNamespace(), []string{"/bin/sh"}, command, false, false)
+			if err != nil {
+				if !mock {
+					return err
+				}
+			}
+			scopedLog.Info("admin password changed on the splunk instance of pod")
+
+			// Get client for Pod and restart splunk instance on pod
+			shClient := mgr.getClient(i)
+			err = shClient.RestartSplunk()
+			if err != nil {
+				return err
+			}
+			scopedLog.Info("Restarted Splunk")
+
+			// Set the adminSecretChanged changed flag to true
+			if i < int32(len(mgr.cr.Status.AdminSecretChanged)) {
+				mgr.cr.Status.AdminSecretChanged[i] = true
+			} else {
+				scopedLog.Info("Appending to AdminSecretChanged")
+				mgr.cr.Status.AdminSecretChanged = append(mgr.cr.Status.AdminSecretChanged, true)
+			}
+
+			// Adding to map of secrets to be synced
+			podSecret, err := splutil.GetSecretFromPod(mgr.c, shPodName, mgr.cr.GetNamespace())
+			if err != nil {
+				return err
+			}
+			mgr.cr.Status.AdminPasswordChangedSecrets[podSecret.GetName()] = true
+			scopedLog.Info("Secret mounted on pod(to be changed) added to map")
+		}
+	}
+
```

Comment:
A comment to explain the reason behind using AdminPasswordChangedSecrets would be helpful

Reply:
Done